src/main/java/com.gigssandbox/UserActivityManager.java

```diff
@@ -0,0 +1,139 @@
+
+import com.gigssandbox.IO.OutputHandler;
+import com.gigssandbox.IO.InputHandler;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+class UserActivityManager {
+    private InputHandler inputHandler;
+    private OutputHandler outputHandler;
+    private UserActivityService userActivityService;
+
+    private String username;
+
+    UserActivityManager(InputHandler inputHandler, OutputHandler outputHandler) {
+        this.inputHandler = inputHandler;
+        this.outputHandler = outputHandler;
+        this.userActivityService = new UserActivityService();
+    }
+
+    void startInteracting() {
+        processUserLoggingIn();
+
+        while (true) {
+            displayPreloaded("choose_action");
+
+            Command currentCommand = getNextCommand();
+            processCommand(currentCommand);
+
+            if (currentCommand == Command.LOG_OUT) {
+                break;
+            }
+        }
+    }
+
+    void processCommand(Command command) {
+        switch (command) {
+            case LOG_OUT:
+                userActivityService.logUserOut(username);
+                displayOnlyTextResponse(command);
+                break;
+            case JOIN_COMMUNITY:
+                userActivityService.addUserToCommunity(username, getParametersForRequest(command));
+                displayPreloaded("joining_community_success");
+                break;
+            case LEAVE_COMMUNITY:
+                userActivityService.removeUserFromCommunity(username);
+                displayOnlyTextResponse(command);
+                break;
+            case GET_GIGS:
+                displayResponse(command, userActivityService.getUpcomingGigs());
+                break;
+            case GET_GIGS_BY_BAND:
+                displayResponse(command, userActivityService.getUpcomingGigsByBand(getParametersForRequest(command).get("band_name")));
+                break;
+            default:
+                break;
+        }
+    }
+
+    void processUserLoggingIn() {
+        String response = "";
+
+        displayPreloaded("need_to_log_in");
+        Map <String, String> currentParameters = getParametersForRequest(Command.LOG_IN);
+
+        while ((response = userActivityService.logUserIn(currentParameters)).equals("incorrect_password")) {
+            displayPreloaded(response);
+            currentParameters.put("password_hash", String.valueOf(getStringParameter().hashCode()));
+        }
+        username = currentParameters.get("username");
+
+        displayPreloaded(response);
+    }
+
+    Command getNextCommand() {
+        int actionCode = getIntParameter();
+        Command currentCommand;
+
+        try {
+            currentCommand = Command.values()[actionCode];
+        } catch (ArrayIndexOutOfBoundsException e) {
+            currentCommand = Command.UNSUPPORTED;
+        }
+        return currentCommand;
+    }
+
+    Map<String, String> getParametersForRequest(Command command) {
```

Comment:
UserActivityManager responsibility to register new users. extract this logic to a new class